DiscordBot.Core/Helpers/TimeZoneHelper.cs

```diff
@@ -22,7 +22,7 @@ public class TimeZoneHelper : ITimezoneHelper
                 var zone = TimeZoneInfo.FindSystemTimeZoneById(timeZone);
                 return zone;
             }
-            catch (TimeZoneNotFoundException)
+            catch (Exception ex) when (ex is TimeZoneNotFoundException || ex is ArgumentNullException)
```

Comment:
It would be better to check if the `timeZone` is null beforehand instead of catching exception